Test/JsonDiffTest.cs

```diff
@@ -0,0 +1,152 @@
+                                 }", 
+                               @"{
+                               'a': {
+                                    '_t': 'a',
+                                     '0': {
+                                           'b': [{
+                                                   'left': 15,
+                                                   'right': 10,
+                                                    'diff': 5
+                                                }]
+                                                },
+                                    '1': {
+                                          'c': [{
+                                                   'left': 25,
+                                                   'right': 20,
+                                                    'diff': 5
+                                               }]
+                                           }
+                                        }
+                                       }" },
+                new object[]
+                { @"{
+                              'number': 10,
+                              'something': {
+                               'inside': 20
+                                }
+                             }", @"{
+                              'number': 5,
+                              'something': {
+                               'inside': 10
+                                }
+                             }", @"{
+                                 'number': [
+                                  {
+                                   'left': 10,
+                                   'right': 5,
+                                   'diff': 5
+                                  }
+                                 ],
+                                 'something': {
+                                    'inside': [
+                                       {
+                                         'left': 20,
+                                         'right': 10,
+                                         'diff': 10
+                                        }
+                                        ]
+                                       }
+                                      }"},
+                new object[]
+                {
+                    @"{
+	                 'number': 10
+                    }", 
+                    @"{
+	                 'number': 10
+                    }",
+                    @"{}"
+                },
+                new object[]
+                {
+                    @"{
+                              'number': 10,
+                              'something': 20
+                             }", @"{
+                              'number': 5,
+                              'something': {
+                               'inside': 10
+                                }
+                             }", @"{
+                                 'number': [{
```

Comment:
Why is the diff object inside an array? Should it not just be `'number': {'left': ...`?

Reply:
If I make it a non-array I am getting some json parse exceptions when I try to parse this string in the tests

Reply:
What I mean is: shouldn't the code generate a single diff object for each property, instead of a list of them? Or are there cases where there may be multiple diff objects created for the same property?

If there is, it might be worth adding a test that explicitly tests that use-case.

Reply:
Yeah it makes more sense to generate just a single diff object instead of arrays. Right now there are no cases for multiple diff objects on a single property